google-cloud-tools-plugin/src/com/google/gct/idea/appengine/cloud/DeploymentArtifactType.java

```diff
@@ -31,11 +32,13 @@
    * Returns the right {@code DeploymentArtifactType} for the given {@code deployPackage}.
    */
   @NotNull
-  public static DeploymentArtifactType typeForPath(@NotNull File deployPackage) {
-    if (deployPackage.getPath().endsWith(".jar")) {
-      return JAR;
-    } else if (deployPackage.getPath().endsWith(".war")) {
-      return WAR;
+  public static DeploymentArtifactType typeForPath(@Nullable File deployPackage) {
```

Comment:
looking at the usages I think it makes sense to let it handle the nulls rather than the callers. The callers are already handling DeploymentArtifactType.UNKNOWN when needed, so my thinking is to just let null just be treated like another unknown type. Let me know if you disagree.